### PR TITLE
docs: update release docs

### PR DESCRIPTION
**Pull request by elchead**
### Proposed change(s)

- we are now exclusively using the `Constellation_CVM` gallery on Azure. 
- add doogfood cluster upgrade to release steps

### Checklist

- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Add labels (e.g., for changelog category)
- [ ] Is PR title adequate for changelog?
- [ ] Link to Milestone

### CODE REVIEW

**Review by malt3 — APPROVED** [review]

**Review by malt3 — APPROVED** [review]
Thanks!

File: dev-docs/workflows/release.md [hunk 1]
```diff
@@ -53,9 +52,20 @@ Releases should be performed using [the automated release pipeline](https://gith
...
    ```
 3. wait for the pipeline to finish
-4. while in editing mode for the release, clear the textbox, select the last minor release and click "Generate release notes".
-5. look over the autogenerated draft release. When fixing the changelog, prioritize updating the PR title/labels/description and regenerating the changelog over fixing things in the final changelog. The changelog should be primarily aimed at users. Rule of thumb: first part of the sentence should describe what changed for the user, second part can describe what has been changed to achieve this.
-6. publish.
+4. upgrade the dogfooding cluster. Note that `upgrade check --update-config` will not yet show the new image. But you can manually set it in the config:
+```sh
+./constellation upgrade check --update-config
+yq eval -i '.image="v2.10.0"' constellation-conf.yaml
+./constellation config fetch-measurements --insecure
```
Comment (derpsteb) [inline dev-docs/workflows/release.md:59]:
```suggestion
./constellation config fetch-measurements
```

is the `insecure` flag really necessary?

File: dev-docs/workflows/release.md [hunk 2]
```diff
@@ -6,7 +6,6 @@ This checklist will prepare `v1.3.0` from `v1.2.0` (minor release) or `v1.3.1` f
 
 1. Search the code for TODOs and FIXMEs that should be resolved before releasing.
 2. [Update titles and labels for all PRs relevant for this release](/dev-docs/conventions.md#pr-conventions) to aid in the [changelog generation](/.github/release.yml).
-3. Check PRs for the label `iam upgrade`. If there is any, update `upgradeRequiresIAMMigration`  in `iamupgradeapply.go`. This ensures the CLI issues a warning on `upgrade apply` to run `iam upgrade apply` before upgrading the cluster.
```
Comment (derpsteb) [inline dev-docs/workflows/release.md:9]: How do we handle this now?

File: dev-docs/workflows/release.md [hunk 3]
```diff
@@ -106,17 +112,24 @@ Currently, this includes: 
...
+
+#### Manual GUI steps
 1. Navigate to [AMI](https://eu-central-1.console.aws.amazon.com/ec2/home?region=eu-central-1#Images:visibility=owned-by-me)
 2. Search for release version "constellation-v1.3.0" and select the AMIs for both variants ("constellation-v1.3.0-aws-sev-snp" and "constellation-v1.3.0-aws-nitro-tpm")
 3. On the "Actions" button (top right) select "Deregister AMI"
 4. Either follow the link on the deletion confirmation leading you to the [Snapshots](https://eu-central-1.console.aws.amazon.com/ec2/home?region=eu-central-1#Snapshots) panel or navigate there yourself
 5. Search for a snapshot by the same name "constellation-v1.3.0" and select it
 6. On the "Actions" button (top right) select "Delete snapshot"
-
-### CDN cache invalidation
```
Comment (derpsteb) [inline dev-docs/workflows/release.md:116]: Why are the cache invalidations not required anymore?

**Review by derpsteb — APPROVED** [review]

File: dev-docs/workflows/release.md (hunk 3 above)
Comment (elchead) [reply dev-docs/workflows/release.md:116]: @malt3  said they are automatically invalidated

**Pull request merged**